plugins/module_utils/support_tunnel.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2023, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+from __future__ import annotations
+
+__metaclass__ = type
+
+from ..module_utils.utils import PayloadMapper
+from .errors import UnexpectedAPIResponse
+from ansible.module_utils.basic import AnsibleModule
+from ..module_utils.client import Client
+from ..module_utils.typed_classes import TypedSupportTunnelToAnsible
+from typing import Any
+
+
+class SupportTunnel(PayloadMapper):
+    def __init__(self, tunnel_open: str):
+        self.tunnel_open = tunnel_open
+
+    @classmethod
+    def from_ansible(cls, ansible_data: Any) -> Any:
+        pass
+
+    @classmethod
+    def from_hypercore(cls, hypercore_data: dict[str, str]) -> SupportTunnel:
+        # There is no None check since get_record is not used (support_tunnel's api behaves different)
+        return cls(
+            tunnel_open=hypercore_data["tunnelOpen"],
+        )
+
+    def to_hypercore(self) -> Any:
+        pass
+
+    def to_ansible(self) -> TypedSupportTunnelToAnsible:
+        return dict(
+            tunnel_open=self.tunnel_open,
+        )
+
+    def __eq__(
+        self, other: object
+    ) -> bool:  # object instead of SupportTunnel to make mypy happy
+        """
+        One support_tunnel is equal to another if it has all attributes exactly the same.
+        This method is used only in tests.
+        """
+        if not isinstance(other, SupportTunnel):
+            return NotImplemented
+        return self.tunnel_open == other.tunnel_open
+
+    @classmethod
+    def check_tunnel_status(cls, client: Client) -> SupportTunnel:
+        response = client.get("/support-api/check")  # type: ignore
+        if response.status == 200:  # get lets 404 through
```

Comment:
When do we get 404? If there is typo in URL, or also is some other case?

Reply:
![image](https://user-images.githubusercontent.com/113340617/217189402-e288153e-0fdc-4865-832a-169145a21a67.png)
In Client we let 200 and 404 through.
Do we need to fix this?

Reply:
@justinc1 In case of support_tunnel only 200 is acceptable, but now after testing all the bad cases I can see only 400 is returned which is already handled in the client, so I will remove the "404 check" from the code. 
@domendobnikar client.py is ok for now